drivers/input/input_chsc6x.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 Nicolas Goualard <nicolas.goualard@sfr.fr>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT seeed_chsc6x
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/input/input.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+
+struct chsc6x_config {
+	struct i2c_dt_spec i2c;
+	const struct gpio_dt_spec int_gpio;
+};
+
+struct chsc6x_data {
+	const struct device *dev;
+	struct k_work work;
+	struct gpio_callback int_gpio_cb;
+};
+
+struct chsc6x_output {
+	uint8_t points;
+	uint16_t x;
+	uint16_t y;
+} __packed;
+
+LOG_MODULE_REGISTER(chsc6x, CONFIG_INPUT_LOG_LEVEL);
+
+static int chsc6x_process(const struct device *dev)
+{
+	uint16_t row, col;
+	bool is_pressed;
+
+	struct chsc6x_output output;
+	const struct chsc6x_config *cfg = dev->config;
+
+	int res = i2c_burst_read_dt(&cfg->i2c, 0, (uint8_t *)&output, sizeof(output));
+
+	if (res < 0) {
+		LOG_ERR("Could not read data %i", res);
+		return -ENODATA;
+	}
+
+	is_pressed = output.points & 0xff;
+	col = sys_be16_to_cpu(output.x) & 0x00ff;
+	row = sys_be16_to_cpu(output.y) & 0x00ff;
+
+	if (is_pressed) {
+		/* These events are generated for the LVGL touch implementation. */
+		input_report_abs(dev, INPUT_ABS_X, col, false, K_FOREVER);
+		input_report_abs(dev, INPUT_ABS_Y, row, false, K_FOREVER);
+		input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
+	} else {
+		/* This event is generated for the LVGL touch implementation. */
+		input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
+	}
+
+	return 0;
+}
+
+static void chsc6x_work_handler(struct k_work *work)
+{
+	struct chsc6x_data *data = CONTAINER_OF(work, struct chsc6x_data, work);
+
+	chsc6x_process(data->dev);
+}
+
+static void chsc6x_isr_handler(const struct device *dev, struct gpio_callback *cb, uint32_t mask)
+{
+	struct chsc6x_data *data = CONTAINER_OF(cb, struct chsc6x_data, int_gpio_cb);
+
+	k_work_submit(&data->work);
+}
+
+static int chsc6x_chip_init(const struct device *dev)
+{
+	const struct chsc6x_config *cfg = dev->config;
+
+	if (!device_is_ready(cfg->i2c.bus)) {
+		LOG_ERR("I2C bus %s not ready", cfg->i2c.bus->name);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int chsc6x_init(const struct device *dev)
+{
+	struct chsc6x_data *data = dev->data;
+
+	data->dev = dev;
+	k_work_init(&data->work, chsc6x_work_handler);
+
+	const struct chsc6x_config *config = dev->config;
+
+	if (!gpio_is_ready_dt(&config->int_gpio)) {
+		LOG_ERR("GPIO port %s not ready", config->int_gpio.port->name);
+		return -EIO;
+	}
+
+	if (gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT) < 0) {
+		LOG_ERR("Could not configure interrupt GPIO pin");
+		return -EIO;
+	}
```

Comment:
```suggestion
	ret = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
	if (ret < 0) {
		LOG_ERR("Could not configure interrupt GPIO pin: %d");
		return ret;
	}
```

^^ I would change this and the other ones to this format, that way the error code is printed and propagated and you may have some more information about how it failed specifically, same for the other ones